app/kudos/models.py

```diff
@@ -544,6 +545,58 @@ def __str__(self):
         """Return the string representation of a model."""
         return f"coupon: {self.coupon} redeemed_by: {self.redeemed_by}"
 
+
+class TokenRequest(SuperModel):
+    """Define the TokenRequest model."""
+
+    name = models.CharField(max_length=255, db_index=True)
+    description = models.TextField(max_length=500, default='')
+    priceFinney = models.IntegerField(default=18)
+    network = models.CharField(max_length=25, db_index=True)
+    artist = models.CharField(max_length=255)
+    platform = models.CharField(max_length=255)
+    to_address = models.CharField(max_length=255)
+    artwork_url = models.CharField(max_length=255)
+    numClonesAllowed = models.IntegerField(default=18)
+    metadata = JSONField(null=True, default=dict, blank=True)
+    tags = ArrayField(models.CharField(max_length=200), blank=True, default=list)
+    approved = models.BooleanField(default=True)
+    processed = models.BooleanField(default=False)
+    profile = models.ForeignKey(
+        'dashboard.Profile', related_name='token_requests', on_delete=models.CASCADE,
+    )
+
+    def __str__(self):
+        """Define the string representation of a conversion rate."""
+        return f"{self.name} on {self.network} on {self.created_on}; approved: {self.approved} "
+
+
+    def mint(self):
+        """Approve / mint this token."""
+        from gas.utils import recommend_min_gas_price_to_confirm_in_time
+        from kudos.management.commands.mint_all_kudos import mint_kudos
+        from kudos.utils import KudosContract
```

Comment:
I assume there is some circular reference funkiness here?

Reply:
yes let me try to remove as many as possible now